packages/hashAlgo/hashAlgo.js

```diff
@@ -0,0 +1,132 @@
+const crypto = require('crypto')
+const EventEmitter = require('events')
+const { invert } = require('lodash')
+
+const preEvents = [
+  {
+    callId: 1,
+    userId: 4545451,
+    method: 'foo.bar1',
+    params: {},
+  },
+  {
+    callId: 2,
+    userId: 45454512,
+    method: 'foo.bar2',
+    params: {},
+  },
+  {
+    callId: 3,
+    userId: 4545453,
+    method: 'foo.bar3',
+    params: {},
+  },
+]
+
+const postEvents = [
+  {
+    callId: 3,
+    result: '',
+  },
+  {
+    callId: 1,
+    result: '',
+  },
+  {
+    callId: 2,
+    result: '',
+  },
+]
+
+const ALGORITHM = 'sha256'
+
+// Format: $<algorithm>$<salt>$<encrypted>
+//
+// http://man7.org/linux/man-pages/man3/crypt.3.html#NOTES
+const ALGORITHM_TO_ID = {
+  md5: '1',
+  sha256: '5',
+  sha512: '6',
+}
+
+const ID_TO_ALGORITHM = invert(ALGORITHM_TO_ID)
+
+const createHash = (data, algorithm = ALGORITHM) =>
+  `$${ALGORITHM_TO_ID[algorithm]}$$${crypto
+    .createHash(algorithm)
+    .update(JSON.stringify(data))
+    .digest('hex')}`
+
+class Audit {
+  constructor(eventEmitter) {
+    this._logs = {}
+    this._lastHash = this._logs.lastHash
```

Comment:
This is `undefined`.

Reply:
I represent `this._logs` as a DB.

Reply:
Even if this is a database, you should not store logs and other data in the same namespace.